Enumerables.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      index = i
+      yield(self[i], index)
+      i += 1
+    end
+    self
+  end
+
+  def my_select(&block)
+    result = []
+    my_each do |element|
+      result << element if block.call(element) == true
+    end
+    result
+  end
+
+  def my_all?(&block)
+    my_each do |element|
+      return false unless block_given? == true && block.call(element) == true
+    end
+    true
+  end
+
+  def my_any?(&block)
+    my_each do |element|
+    if block_given? == true
+      return true if block.call(element) == true
+    else
+      return true if (element) == true
+    end
+    end
+    false
+  end
+
+  def my_none?(&block)
+    my_each do |element|
+    if block_given? == true
+      return false unless block.call(element) == false
+    else
+      return false unless (element) == false
+    end
+    end
+    true
+  end
+
+  def my_count(&block)
+    count = 0
+    my_each do |element|
+      count += 1 if block.call(element) == true
```

Comment:
`my_count` seems not working, you need to work on it, got two errors stated below while running the method.
```
     NoMethodError:
       undefined method `call' for nil:NilClass
```
```
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```